packages/babel-types/src/modifications/flow/removeTypeDuplicates.ts

```diff
@@ -1,17 +1,24 @@
-// @flow
 import {
   isAnyTypeAnnotation,
   isGenericTypeAnnotation,
   isUnionTypeAnnotation,
   isFlowBaseAnnotation,
+  isIdentifier,
 } from "../../validators/generated";
+import type * as t from "../../types";
+
+function getQualificationName(node: t.GenericTypeAnnotation["id"]) {
+  return isIdentifier(node)
+    ? node.name
+    : `${node.id.name}.${getQualificationName(node.qualification)}`;
+}
```

Comment:
Uh is this a bugfix caught by TS?

Btw, nit but the function name should be `getQualifiedName`.

Reply:
yes, the first one I actually found with TS :)
updating the name